### PR TITLE
Self chat general script + Wizard self chat example world

### PR DESCRIPTION
**Pull request by jaseweston**

### CODE REVIEW

**Comment by jaseweston** [comment]
e.g. rn run with:

py parlai/scripts/self_chat.py -mf /checkpoint/jase/projects/selfchats/retrieval/self2c_wiz_neg/5cb_jobid=2/model -t wizard_of_wikipedia:selfchat

File: parlai/core/agents.py [hunk 1]
```diff
@@ -114,6 +114,13 @@ def save(self, path=None):
         """
         pass
 
+    def clone(self):
+        """
+        Make a shared copy of this agent.
+        Should be the same as using create_agent_from_shared(.), but slightly easier.
+        """
+        return type(self)(self.opt, self.share())
```
Comment (stephenroller) [inline parlai/core/agents.py]: Love it

File: parlai/core/torch_agent.py [hunk 2]
```diff
@@ -1734,7 +1734,10 @@ def batch_act(self, observations):
         """
         batch_size = len(observations)
         # initialize a list of replies with this agent's id
-        batch_reply = [Message({'id': self.getID()}) for _ in range(batch_size)]
+        batch_reply = [
+            Message({'id': self.getID(), 'episode_done': False})
```
Comment (stephenroller) [inline parlai/core/torch_agent.py]: I’m worried about this. Will this cause the teacher metrics to get messed up?

File: parlai/core/worlds.py [hunk 3]
```diff
@@ -1150,7 +1150,7 @@ def _get_task_world(opt, user_agents, default_world=None):
         try:
             my_module = importlib.import_module(module_name)
             world_class = getattr(my_module, world_name)
-        except Exception:
+        except AttributeError:
```
Comment (stephenroller) [inline parlai/core/worlds.py]: `except (ModuleNotFoundError, AttributeError)`

File: parlai/scripts/self_chat.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+import random
+
+
+def setup_args(parser=None):
+    if parser is None:
+        parser = ParlaiParser(True, True, 'Self chat with a model')
+    parser.add_argument('-d', '--display-examples', type='bool', default=True)
+    parser.add_argument(
+        '--display-prettify',
```
Comment (stephenroller) [inline parlai/scripts/self_chat.py]: ?

File: parlai/scripts/self_chat.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            print_parser = None
+    if isinstance(opt, ParlaiParser):
+        print('[ Deprecated Warning: self_chat should be passed opt not Parser ]')
+        opt = opt.parse_args()
+
+    # Create models
+    agent1 = create_agent(opt, requireModelExists=True)
+    agent2 = agent1.clone()
+    if hasattr(agent2, 'id'):
+        agent2.id = agent2.id + "2"
```
Comment (stephenroller) [inline parlai/scripts/self_chat.py]: Nice

File: parlai/scripts/self_chat.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        if opt.get('display_examples'):
+            print("---")
+            print(world.display())
+        if world.epoch_done():
+            print("EPOCH DONE")
+            break
+
+
+if __name__ == '__main__':
+    random.seed(42)
```
Comment (stephenroller) [inline parlai/scripts/self_chat.py]: Can we make seed an argument? I find it infuriating to not be able to easily switch up the personas etc.

File: parlai/scripts/self_chat.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        help='Do not display these fields',
+    )
+    parser.add_argument(
+        '-it',
+        '--interactive-task',
+        type='bool',
+        default=True,
+        help='Create interactive version of task',
+    )
+    parser.set_defaults(interactive_mode=True, task='self_chat')
```
Comment (stephenroller) [inline parlai/scripts/self_chat.py]: Do you need that task=self_chat?

Comment (jaseweston) [reply parlai/scripts/self_chat.py]: it makes it so that it can default do something, otherwise u need to specify the right task..

**Pull request closed without merging**